NtupleProducer/plugins/bRegressionProducer.cc

```diff
@@ -0,0 +1,567 @@
+    Jet_energyRing_dR2_ch_Jet_e = 0.;
+    Jet_energyRing_dR3_ch_Jet_e = 0.;
+    Jet_energyRing_dR4_ch_Jet_e = 0.;
+    Jet_energyRing_dR0_mu_Jet_e = 0.;
+    Jet_energyRing_dR1_mu_Jet_e = 0.;
+    Jet_energyRing_dR2_mu_Jet_e = 0.;
+    Jet_energyRing_dR3_mu_Jet_e = 0.;
+    Jet_energyRing_dR4_mu_Jet_e = 0.;
+    Jet_numDaughters_pt03 = 0;
+
+    Jet_chHEF = 0.;//implement from here
+    Jet_chEmEF = 0.;
+    Jet_leptonPtRelInv = 0.;
+    isEle = 0;
+    isMu = 0;
+    isOther = 0;
+    Jet_mass = 0.;
+    Jet_withPtd = 0.;
+
+}//end InitJet
+
+void bRegressionProducer::SetNNVectorVar(){
+
+    NNvectorVar_.push_back(Jet_pt) ;//0
+    NNvectorVar_.push_back(Jet_eta) ;
+    NNvectorVar_.push_back(rho) ;
+    NNvectorVar_.push_back(Jet_mt) ;
+    NNvectorVar_.push_back(Jet_leadTrackPt) ;
+    NNvectorVar_.push_back(Jet_leptonPtRel) ;//5
+    NNvectorVar_.push_back(Jet_leptonDeltaR) ;
+    NNvectorVar_.push_back(Jet_neHEF) ;
+    NNvectorVar_.push_back(Jet_neEmEF) ;
+    NNvectorVar_.push_back(Jet_vtxPt) ;
+    NNvectorVar_.push_back(Jet_vtxMass) ;//10
+    NNvectorVar_.push_back(Jet_vtx3dL) ;
+    NNvectorVar_.push_back(Jet_vtxNtrk) ;
+    NNvectorVar_.push_back(Jet_vtx3deL) ;
+    NNvectorVar_.push_back(Jet_numDaughters_pt03) ;//this variable has changed order, in bdt it was last, check why
+    NNvectorVar_.push_back(Jet_energyRing_dR0_em_Jet_e) ;//15
+    NNvectorVar_.push_back(Jet_energyRing_dR1_em_Jet_e) ;
+    NNvectorVar_.push_back(Jet_energyRing_dR2_em_Jet_e) ;
+    NNvectorVar_.push_back(Jet_energyRing_dR3_em_Jet_e) ;
+    NNvectorVar_.push_back(Jet_energyRing_dR4_em_Jet_e) ;
+    NNvectorVar_.push_back(Jet_energyRing_dR0_neut_Jet_e) ;//20
+    NNvectorVar_.push_back(Jet_energyRing_dR1_neut_Jet_e) ;
+    NNvectorVar_.push_back(Jet_energyRing_dR2_neut_Jet_e) ;
+    NNvectorVar_.push_back(Jet_energyRing_dR3_neut_Jet_e) ;
+    NNvectorVar_.push_back(Jet_energyRing_dR4_neut_Jet_e) ;
+    NNvectorVar_.push_back(Jet_energyRing_dR0_ch_Jet_e) ;//25
+    NNvectorVar_.push_back(Jet_energyRing_dR1_ch_Jet_e) ;
+    NNvectorVar_.push_back(Jet_energyRing_dR2_ch_Jet_e) ;
+    NNvectorVar_.push_back(Jet_energyRing_dR3_ch_Jet_e) ;
+    NNvectorVar_.push_back(Jet_energyRing_dR4_ch_Jet_e) ;
+    NNvectorVar_.push_back(Jet_energyRing_dR0_mu_Jet_e) ;//30
+    NNvectorVar_.push_back(Jet_energyRing_dR1_mu_Jet_e) ;
+    NNvectorVar_.push_back(Jet_energyRing_dR2_mu_Jet_e) ;
+    NNvectorVar_.push_back(Jet_energyRing_dR3_mu_Jet_e) ;
+    NNvectorVar_.push_back(Jet_energyRing_dR4_mu_Jet_e) ;
+    NNvectorVar_.push_back(Jet_chHEF);//35
+    NNvectorVar_.push_back(Jet_chEmEF);
+    NNvectorVar_.push_back(Jet_leptonPtRelInv);
+    NNvectorVar_.push_back(isEle);
+    NNvectorVar_.push_back(isMu);
+    NNvectorVar_.push_back(isOther);//40
+    NNvectorVar_.push_back(Jet_mass);
+    NNvectorVar_.push_back(Jet_withPtd);
+
+}
+    
+std::vector<float> bRegressionProducer::EvaluateNN(){
+    tensorflow::Tensor input(tensorflow::DT_FLOAT, {1,43});
```

Comment:
What is the meaning of ```{1,43}```? A comment could be useful.

Reply:
it's the input vector format. I'll add a comment (useful indeed in case variables get added/removed)